core/src/main/java/org/apache/calcite/plan/volcano/VolcanoPlanner.java

```diff
@@ -392,35 +382,24 @@ public RelSet getSet(RelNode rel) {
     this.mapRel2Subset.clear();
     this.relImportances.clear();
     this.ruleQueue.clear();
-    this.ruleNames.clear();
     this.materializations.clear();
     this.latticeByName.clear();
     this.provenanceMap.clear();
   }
 
   public List<RelOptRule> getRules() {
-    return ImmutableList.copyOf(ruleSet);
+    return ImmutableList.copyOf(mapDescToRule.values());
   }
 
   public boolean addRule(RelOptRule rule) {
```

Comment:
If we choose to add this method to the superclass (instead of having `mapRuleDescription`) then here we can ovverride.

Reply:
@zabetak I have addressed you comments. Thanks.